### PR TITLE
test(SwingSet): Tooling to update xsnap hashes

### PR DESCRIPTION
**Pull request by kriskowal**
This change will make it easier going forward to update the hash consistency checks for xsnap snapshots.
The oracle hashes need to be updated whenever anything in the xsnap SwingSet bootstrap changes.
The test includes instructions for its usage which will appear in test output only when the test fails.

### CODE REVIEW

**Comment by warner** [comment]
Good plan. Maybe store them as an object a meaningful name for each of the three, instead of an array?

**Comment by kriskowal** [comment]
> Good plan. Maybe store them as an object a meaningful name for each of the three, instead of an array?

My first sketch gave names:

1. `"initial"`
2. `"bootstrap"`
3. `"after-adding-global-property-named-x-that-was-a-empty-hardened-object"`

Send halp.

**Comment by michaelfig** [comment]
> 1. `"initial"`
> 2. `"bootstrap"`
> 3. `"after-adding-global-property-named-x-that-was-a-empty-hardened-object"`

3. `"sanity-test"`

> Send halp.

I heard your plea.

File: packages/SwingSet/test/test-xsnap-store.js [hunk 1]
```diff
@@ -149,12 +151,15 @@ test('XS + SES snapshots are long-term deterministic', async t => {
...
-  const h1 = await store.save(vat.snapshot);
-  t.is(
-    h1,
-    '5074bc800f9d6f1e1821c2a50214f9ccdf4d45507e44409455f3da08d937e8a1',
-    'initial snapshot',
+  const hashesPath = url.fileURLToPath(
+    new URL('xsnap-store-hashes.json', import.meta.url),
   );
+  let hashesBytes = await fs.promises.readFile(hashesPath);
+  let hashesText = new TextDecoder().decode(hashesBytes);
```
Comment (dckc) [inline packages/SwingSet/test/test-xsnap-store.js:158]:
```suggestion
  let hashesText = await fs.promises.readFile(hashesPath, 'utf-8');
```

**Review by dckc — APPROVED** [review]
fs.promise.readFile includes text decoding if you include the 3rd arg

**Comment by turadg** [comment]
This is helpful tooling. Maybe this is naive, but why not use Ava's native snapshotting? https://github.com/Agoric/agoric-sdk/pull/4400

**Pull request merged**